### ISSUE TITLE
The `list-item-indent` rule is buggy when it comes to checkboxes

### ISSUE THREAD

**Issue by sodatea** [wooorm/remark-lint#42]
```
╭─soda  ~/Tests/remark-lint
╰─$ npm list -g --depth=0 remark remark-lint
/Users/soda/.nvm/versions/node/v4.2.6/lib
+-- remark@3.2.2
`-- remark-lint@2.2.1

╭─soda  ~/Tests/remark-lint
╰─$ cat test.md
- [ ] Item 1
- [x] Item 2
- [ ] Item 3
╭─soda  ~/Tests/remark-lint
╰─$ cat .remarkrc
{
    "plugins": {
        "lint": {
            "reset": true,
            "list-item-indent": true
        }
    },
    "settings": {
        "gfm": true
    }
}
╭─soda  ~/Tests/remark-lint
╰─$ remark test.md -u remark-lint -c .remarkrc
-   [ ] Item 1
-   [x] Item 2
-   [ ] Item 3
test.md
        1:7  warning  Incorrect list-item indent: add 2 spaces  list-item-indent
        1:7  warning  Incorrect list-item indent: add 2 spaces  list-item-indent
        2:7  warning  Incorrect list-item indent: add 2 spaces  list-item-indent
        2:7  warning  Incorrect list-item indent: add 2 spaces  list-item-indent
        3:7  warning  Incorrect list-item indent: add 2 spaces  list-item-indent
        3:7  warning  Incorrect list-item indent: add 2 spaces  list-item-indent

⚠ 6 warnings
```

**Comment by sodatea** [comment]
Actually it is broken for *any* lists……

Maybe I misunderstood this rule's meaning?

**Comment by wooorm** [comment]
Well, first of all, remark detects the `.remarkrc` file and automatically loads the used plug-ins, namely, `remark-lint`, so you don’t need to specify it on the CLI too (that’s why you get double warnings).

The default style suggested by `list-item-indent` is to `"tab-size"` indentation, which is why you see the warnings (as it detects an indentation of `2`, not `4`). See the [docs on how to change that behviour](https://github.com/wooorm/remark-lint/blob/master/doc/rules.md#list-item-indent).

**Comment by sodatea** [comment]
1. I have reset this rule to `'space'` (there was a typo in the original post, but I've fixed that and the error remains)
2. I typed only one space, but remark detected 2, this is confusing

**Comment by sodatea** [comment]
Oh I see. It's the `-u remark-lint` part in the command that causes this weird behavior.

**Comment by wooorm** [comment]
See the first paragraph of my previous post, you must remove `-u remark-lint`.

**Comment by wooorm** [comment]
Yup, it causes duplicate remark-lint, the first with options the second without

**Comment by sodatea** [comment]
But it's copied from README.md

> Use remark-lint together with remark:
> 
> ```bash
> npm install --global remark remark-lint
> ```
> 
> Let’s say `example.md` looks as follows:
> 
> ```md
> * Hello
> 
> [World][]
> ```
> 
> Then, to run **remark-lint** on `example.md`:
> 
> ```bash
> remark example.md -u remark-lint
> 

The documentation needs to be updated.

**Comment by wooorm** [comment]
No, the usage section does not contain a remarkrc file.
The remarkrc example later is a different example, showing different settings and different results.
